addons/account/models/account_account.py

```diff
@@ -858,6 +1089,262 @@ def get_import_templates(self):
+                user_lock_date = company._get_user_fiscal_lock_date()
+                if user_lock_date and (locked_items := self.env['account.move.line'].sudo().search([
+                    ('account_id', 'in', (accounts - account_to_merge_into).ids),
+                    ('date', '<=', user_lock_date),
+                ])):
+                    if not account_to_merge_into and len(locked_items.account_id) == 1:
+                        account_to_merge_into = locked_items.account_id
+                    else:
+                        merge_issues[company]['locked_entries'] = locked_items.move_id
+                        merge_issues[company]['lock_date'] = user_lock_date
+
+        if merge_issues:
+            error_message = _("The merge could not proceed. The following issues were noted:") + "\n\n"
+            for company, issues in merge_issues:
+                error_message += _("Company %s:", company.name) + "\n"
+                if locked_entries := issues.get('locked_entries'):
+                    error_message += _("- The following locked journal entries would be modified (lock date: %s)", issues['lock_date'])
+                    error_message += "\n".join(f"    - {entry.name}" for entry in locked_entries)
+
+        # Choose account to merge into. If one account already contains locked entries, merge the other accounts into it.
+        # Otherwise, we choose the first account to be the one we merge other accounts into.
+        if not account_to_merge_into:
+            account_to_merge_into = self[0]
+        accounts_to_remove = self - account_to_merge_into
+
+        # Step 2: Ask the user for confirmation
+        if not self.env.context.get('account_merge_confirm'):
+            msg = _("Are you sure? This will perform the following operations:\n")
+            for account in accounts_to_remove:
+                msg += _(
+                    "- %(account_1)s (company: %(companies_1)s) will be merged into %(account_2)s (company: %(companies_2)s)\n",
+                    account_1=account.with_company(account.company_ids[:1]).display_name,
+                    companies_1=",".join(account.company_ids.mapped('name')),
+                    account_2=account_to_merge_into.with_company(account_to_merge_into.company_ids[:1]).display_name,
+                    companies_2=",".join(account_to_merge_into.company_ids.mapped('name')),
+                )
+            if is_irreversible:
+                msg += _(
+                    "This cannot be undone because you are merging accounts belonging to the same company.\n"
+                    "After merging, we won't be able to separate journal entries based on which account they originally referenced."
+                )
+            action = self.env['ir.actions.actions']._for_xml_id('account.action_merge_accounts')
+            raise RedirectWarning(msg, action, _("Merge"), additional_context={**self.env.context, 'account_merge_confirm': True})
```

Comment:
Is this enough to get the right `self` when confirming? (I supposed based on the `active_ids`?)

Reply:
Yes exactly, the active_ids are copied from self.env.context :+1: